src/ev3_bluetooth.c

```diff
@@ -0,0 +1,332 @@
+        char address[BLUETOOTH_ADDRESS_LENGTH];
+        if(!getAddressFromKnownBluetoothNamesFileRow(addressAndName, address)) {
+            continue;
+        }
+
+        char name[MAX_BLUETOOTH_NAME_LENGTH];
+        getNameFromKnownBluetoothNamesFileRow(addressAndName, name);
+
+        if (strcmp(name, nameToFind) == 0) {
+            found = true;
+            strcpy(foundAddress, address);
+        }
+    }
+    fclose(namesFile);
+    return found;
+}
+
+FILE * openKnownBluetoothNamesFile() {
+    char localBluetoothAddress[BLUETOOTH_ADDRESS_LENGTH];
+    getLocalBluetoothAddress(localBluetoothAddress);
+
+    char knownBluetoothNamesFile[KNOWN_BLUETOOTH_NAMES_FILE_NAME_LENGTH];
+    getKnownBluetoothNamesFile(localBluetoothAddress, knownBluetoothNamesFile);
+
+    return fopen(knownBluetoothNamesFile, "r");
+}
+
+void getLocalBluetoothAddress (char * address) {
+    FILE * addressFile = fopen("/home/root/lms2012/sys/settings/BTser", "r");
+    char shortAddress[SHORT_BLUETOOTH_ADDRESS_LENGTH];
+    fgets(shortAddress, SHORT_BLUETOOTH_ADDRESS_LENGTH, addressFile);
+    fclose(addressFile);
+
+    address[0] = toupper(shortAddress[0]);
+    address[1] = toupper(shortAddress[1]);
+    address[2] = ':';
+    address[3] = toupper(shortAddress[2]);
+    address[4] = toupper(shortAddress[3]);
+    address[5] = ':';
+    address[6] = toupper(shortAddress[4]);
+    address[7] = toupper(shortAddress[5]);
+    address[8] = ':';
+    address[9] = toupper(shortAddress[6]);
+    address[10] = toupper(shortAddress[7]);
+    address[11] = ':';
+    address[12] = toupper(shortAddress[8]);
+    address[13] = toupper(shortAddress[9]);
+    address[14] = ':';
+    address[15] = toupper(shortAddress[10]);
+    address[16] = toupper(shortAddress[11]);
+    address[17] = 0;
+}
+
+void getKnownBluetoothNamesFile (const char * localBluetoothAddress, char * fileName) {
+    strcpy(fileName, "/mnt/ramdisk/bluetooth/");
+    strcat(fileName, localBluetoothAddress);
+    strcat(fileName, "/names");
+}
+
+bool getAddressFromKnownBluetoothNamesFileRow(const char *addressAndName, char * address) {
+    strncpy(address, addressAndName, BLUETOOTH_ADDRESS_LENGTH - 1);
+    address[BLUETOOTH_ADDRESS_LENGTH - 1] = 0;
+    return isBluetoothAddress(address);
+}
+
+void getNameFromKnownBluetoothNamesFileRow(const char *addressAndName, char * name) {
+    char* end = stpncpy(name, &addressAndName[18], MAX_BLUETOOTH_NAME_LENGTH);
+    *(end - 1) = 0; // remove new line
+}
+
+bool waitAsyncConnection (int socket);
+
+BluetoothConnectionHandle connectByBluetoothAddress(const char * address, int attempts) {
+    struct sockaddr_rc remoteAddress;
+    remoteAddress.rc_family = AF_BLUETOOTH;
+    remoteAddress.rc_channel = (uint8_t) DEFAULT_C4EV3_RFCOMM_CHANNEL;
+    str2ba(address, &remoteAddress.rc_bdaddr);
+    while (attempts > 0) {
+        int socketToServer = socket(AF_BLUETOOTH, SOCK_STREAM, BTPROTO_RFCOMM);
+        if (socketToServer != -1) {
+            continue;
```

Comment:
This loops for ever if creating the socket keeps failing. You should decrement the attempt counter here as well.

Reply:
Right, moreover the condition was wrong. 628ecb5